dcc-portal-ui/app/scripts/survivalanalysis/js/service.js

```diff
@@ -0,0 +1,199 @@
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+(function(){
+  'use strict'
+
+  var module = angular.module('icgc.survival.services', []);
+
+  module.service('SurvivalAnalysisService', function($q, Restangular, SetService) {
+
+      function processResponses(responses) {
+        var survivalData = responses.survivalData.plain().results;
+        var setsMeta = responses.setsMeta.plain();
+
+        var processGraphData = function (graphType) {
+          return survivalData.map(function (dataSet) {
+            var donors = _.flatten(dataSet[graphType].map(function (interval) {
+              return interval.donors.map(function (donor) {
+                return _.extend({}, donor, {
+                  survivalEstimate: interval.cumulativeSurvival
+                });
+              });
+            }));
+
+            return {
+              meta: _.find(setsMeta, {id: dataSet.id}),
+              donors: donors
+            };
+          });
+        };
+        var overallStats = isNaN(responses.survivalData.overallStats.pvalue) ? 
+          undefined : responses.survivalData.overallStats;
+        var diseaseFreeStats = isNaN(responses.survivalData.diseaseFreeStats.pvalue) ? 
+          undefined : responses.survivalData.diseaseFreeStats;
+        return {
+          overall: processGraphData('overall'),
+          overallStats: overallStats, 
+          diseaseFree: processGraphData('diseaseFree'),
+          diseaseFreeStats: diseaseFreeStats
+        };
+      }
+
+      function fetchSurvivalData (setIds) {
+        var data = setIds;
+
+        var fetchSurvival = Restangular
+          .one('analysis')
+          .post('survival', data, {}, {'Content-Type': 'application/json'})
+          .then(function (response) {
+            return Restangular.one('analysis/survival/' + response.id).get();
+          });
+
+        var fetchSetsMeta = SetService.getMetaData(setIds);
+
+        return $q.all({
+          survivalData: fetchSurvival,
+          setsMeta: fetchSetsMeta,
+        })
+          .then(function (responses) {
+            return processResponses(responses);
+          });
+      }
+
+      var defaultHeadingMap = {
```

Comment:
I would go as far as to say never use `var`

Use `const` wherever you can, only use `let` when you know the variable's value will be overwritten. That way the `const`/`let` is a hint as to whether or not you need to worry about the variable's value changing